mflowgen/Tile_MemCore/custom-init/outputs/pin-assignments.tcl

```diff
@@ -71,6 +72,18 @@ for {set i 0} {$i < [llength $tile_id]} {incr i} {
   }
 }
 
-editPin -pin $id_ports -start [list 0 [expr {$height - 14}]] -end [list 0 [expr {$height - 5}]] -side LEFT -spreadType RANGE -spreadDirection clockwise -layer M6
-
-
+editPin -pin $id_ports -start [list 0 [expr {$height - 14}]] -end [list 0 [expr {$height - 5}]] -side LEFT -spreadType RANGE -spreadDirection clockwise -layer $tile_id_layer
+
+# Add blockage in area near tile_id pins so we can route these at the top level
+set tile_id_y_coords [get_property [get_ports {*tile_id* hi* lo*}] y_coordinate]
+set tile_id_y_coords [lsort -real $tile_id_y_coords]
+set tile_id_min_y [lindex $tile_id_y_coords 0]
+set tile_id_max_y [lindex $tile_id_y_coords end]
+selectIOPin *tile_id*
+set tile_id_max_x [dbGet [dbGet selected -i 0].pins.allShapes.shapes.rect_urx]
+deselectAll
+
+createRouteBlk -name tile_id_rb -layer $tile_id_layer -box [list 0 $tile_id_min_y $tile_id_max_x $tile_id_max_y]
+#create route blockage on other side of tile aligned with tile id pins
+set horiz_pitch [dbGet top.fPlan.coreSite.size_x]
+createRouteBlk -name tile_id_oppo -layer $tile_id_layer -box [list [expr $width - (5 * $horiz_pitch)] $tile_id_min_y $width $tile_id_max_y]
```

Comment:
So you create a larger x gap on the other side? I'm assuming that 5*horiz_pitch is larger than tile_id_max_x. I guess this makes sense since you need some space beyond just the pin area to the do the routing?

Reply:
Yes. Perhaps this could be smaller, but the point is to ensure that those nets don't get too close to my manual routes and cause DRCs.